src/parser.rs

```diff
@@ -5856,22 +5870,50 @@ mod tests {
             }};
         }
 
-        let dummy_name = ObjectName(vec![Ident::new("dummy_name")]);
+        let dummy_name = SchemaName::new(Ident::new("dummy_name"), None);
         let dummy_authorization = Ident::new("dummy_authorization");
 
         test_parse_schema_name!(
             format!("{dummy_name}"),
-            SchemaName::Simple(dummy_name.clone())
+            SchemaNameClause::Simple(dummy_name.clone())
         );
 
         test_parse_schema_name!(
             format!("AUTHORIZATION {dummy_authorization}"),
-            SchemaName::UnnamedAuthorization(dummy_authorization.clone()),
+            SchemaNameClause::UnnamedAuthorization(dummy_authorization.clone()),
         );
         test_parse_schema_name!(
             format!("{dummy_name} AUTHORIZATION {dummy_authorization}"),
-            SchemaName::NamedAuthorization(dummy_name.clone(), dummy_authorization.clone()),
+            SchemaNameClause::NamedAuthorization(dummy_name.clone(), dummy_authorization.clone()),
+        );
+
+        let dummy_qualified_name =
+            SchemaName::new(Ident::new("dummy_name"), Some(Ident::new("potato")));
+
+        test_parse_schema_name!(
+            format!("{dummy_qualified_name}"),
+            SchemaNameClause::Simple(dummy_qualified_name.clone())
         );
+
+        test_parse_schema_name!(
+            format!("AUTHORIZATION {dummy_authorization}"),
+            SchemaNameClause::UnnamedAuthorization(dummy_authorization.clone()),
+        );
+        test_parse_schema_name!(
+            format!("{dummy_qualified_name} AUTHORIZATION {dummy_authorization}"),
+            SchemaNameClause::NamedAuthorization(
+                dummy_qualified_name.clone(),
+                dummy_authorization.clone()
+            ),
+        );
+    }
+
+    #[test]
+    #[should_panic = "Expected schema name with no more than two qualifications, but found a period after the second name"]
```

Comment:
Like for example, I think in many cases this parser library is fairly permissive -- so unless there is some problem accepting the full database.schema.table type qualified name I would be hesitant to remove support